### PR TITLE
networking: Fix IP Settings dialog

### PR DESCRIPTION
**Pull request by mvollmer**
Initializion of the dialog was wrong for ignore-auto-dns,
ignore-auto-routes, and dns-search.  Taking values out of the dialog
was wrong for dns-search.

### CODE REVIEW

**Comment by martinpitt** [comment]
Moving to draft for now, as this still needs a test case and has some failures. But labeling release-blocker.

**Comment by martinpitt** [comment]
@mvollmer : On your next push, can you please add a ref to https://bugzilla.redhat.com/show_bug.cgi?id=2023521 and https://bugzilla.redhat.com/show_bug.cgi?id=2056446 to the commit message?

File: pkg/networkmanager/ip-settings.jsx [hunk 1]
```diff
@@ -262,7 +262,7 @@ export const IpSettingsDialog = ({ topic, connection, dev, setIsOpen, settings }
                                 <Button variant="secondary"
                                         isDisabled={!canHaveExtra}
                                         onClick={() => setDnsSearch([...dnsSearch, ""])}
-                                        id={idPrefix + "-dnf-search-add"}
+                                        id={idPrefix + "-dns-search-add"}
```
Comment (mvollmer) [inline pkg/networkmanager/ip-settings.jsx:265]: This is my favorite typo so far. :-)

**Comment by mvollmer** [comment]
> On your next push, can you please add a ref to https://bugzilla.redhat.com/show_bug.cgi?id=2023521 and https://bugzilla.redhat.com/show_bug.cgi?id=2056446 to the commit message?

2056446 is not related, no?

**Comment by martinpitt** [comment]
@mvollmer : Ah right, sorry; they all sound so similar. So the auto-reverting on/off button is still open then.

**Comment by mvollmer** [comment]
> @mvollmer : Ah right, sorry; they all sound so similar. So the auto-reverting on/off button is still open then.

That is fixed by https://github.com/cockpit-project/cockpit/pull/17036.

**Comment by martinpitt** [comment]
I see -- for that PR (commit 5b4ed2fce2b4) I added https://bugzilla.redhat.com/show_bug.cgi?id=2056362 and https://bugzilla.redhat.com/show_bug.cgi?id=2056386 . So should we mark  https://bugzilla.redhat.com/show_bug.cgi?id=2056446 as another duplicate then?

**Review by martinpitt — APPROVED** [review]
Danke!

**Pull request merged**